server/engine/src/main/java/org/eclipse/lsp/cobol/core/model/tree/logic/CICSTranslatorProcessor.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (c) 2022 Broadcom.
+ * The term "Broadcom" refers to Broadcom Inc. and/or its subsidiaries.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *    Broadcom, Inc. - initial API and implementation
+ *
+ */
+package org.eclipse.lsp.cobol.core.model.tree.logic;
+
+import lombok.AllArgsConstructor;
+import org.eclipse.lsp.cobol.core.engine.processor.ProcessingContext;
+import org.eclipse.lsp.cobol.core.engine.processor.Processor;
+import org.eclipse.lsp.cobol.core.model.ErrorSeverity;
+import org.eclipse.lsp.cobol.core.model.ErrorSource;
+import org.eclipse.lsp.cobol.core.model.SyntaxError;
+import org.eclipse.lsp.cobol.core.model.tree.CICSTranslatorNode;
+import org.eclipse.lsp.cobol.service.AnalysisConfig;
+
+/** Handles logic to enable disable CICS Translator options */
+@AllArgsConstructor
+public class CICSTranslatorProcessor implements Processor<CICSTranslatorNode> {
+
+  final AnalysisConfig analysisConfig;
+  /**
+   * Handles logic to enable disable CICS Translator options based on settings.
+   *
+   * @param cicsTranslatorNode {@link CICSTranslatorNode}
+   * @param processingContext {@link ProcessingContext}
+   */
+  @Override
+  public void accept(CICSTranslatorNode cicsTranslatorNode, ProcessingContext processingContext) {
+    if (!analysisConfig.isCicsTranslatorEnabled()) {
+      processingContext
+          .getErrors()
+          .add(
+              SyntaxError.syntaxError()
+                  .locality(cicsTranslatorNode.getLocality())
+                  .severity(ErrorSeverity.ERROR)
+                  .errorSource(ErrorSource.PARSING)
+                  .suggestion("Enable CICS translator")
```

Comment:
@VitGottwald  @zeibura @zimlu02 Any suggestion for the error message when cics is disabled but we find CICS translator options

Reply:
maybe "Enable CICS translator in the settings" ?

Reply:
CICS is disabled where?

Reply:
To enable CICS compilation in the cobol program, **CICS translator** compiler directives were used. But with the latest cobol version, this support is inbuilt and doesn't require these directives. Some customers want to support these statements while others want to throw errors on them.

Reply:
Updated to - "Enable CICS translator in the settings"

Reply:
![image](https://user-images.githubusercontent.com/69206564/195133900-74db7a15-a1d9-417a-b6c7-af3a54b1e844.png)

Reply:
Yes. If we disable the CICS translator from the extension settings, we start to show an error, if the program used the CICS translator within the cobol code. 
By default, it is set to true. 
Basically (as I understand), the CICS translator depends on the cobol compiler. If a company/organization is using an old cobol compiler they would need to have these to support CICS.
While the newer compiler doesn't need these and we need to show errors on those statements.

Reply:
So I'm not sure "translator" is the best term to use in public facing texts. How about "Enable CICS support"?